filters/auth/authclient.go

```diff
@@ -122,12 +125,16 @@ func jsonPost(u *url.URL, auth string, doc *tokenIntrospectionInfo, client *http
 
 // doClonedGet requests url with the same headers and query as the
 // incoming request and returns with http statusCode and error.
-func doClonedGet(u *url.URL, client *http.Client, incoming *http.Request) (int, error) {
+func doClonedGet(u *url.URL, client *http.Client, incoming *http.Request, tracer opentracing.Tracer,
+	parentSpan opentracing.Span, childSpanName string) (int, error) {
+	span := tracer.StartSpan(childSpanName, opentracing.ChildOf(parentSpan.Context()))
+	defer span.Finish()
```

Comment:
I think it would be nicer to pass 'span` and move the 3 lines of code to `getWebhook`.

Reply:
the tracer can be a field of the authClient. The doClonedGet could be a method if the *authClient. Since we alwyas use the webhookSpanName constant, it is not required as a parameter. This way doClonedGet would have only the following parameters: incoming and parentSpan.

Reply:
sry: i see that the tracer cannot be the field of the client without bigger changes, so let's not do that. But the incoming request, the tracer and the parent span name are all coming from the filter context, so why not just passing down the filter context?